core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -712,10 +708,13 @@ class ReplicaManager(val config: KafkaConfig,
         } catch {
           case e@(_: InvalidTopicException |
                   _: LogDirNotFoundException |
-                  _: ReplicaNotAvailableException |
                   _: KafkaStorageException) =>
-            warn("Unable to alter log dirs for %s".format(topicPartition), e)
+            warn(s"Unable to alter log dirs for $topicPartition", e)
             (topicPartition, Errors.forException(e))
+          case e: NotLeaderOrFollowerException =>
+            warn(s"Unable to alter log dirs for $topicPartition", e)
+            // Retaining REPLICA_NOT_AVAILABLE exception for ALTER_REPLICA_LOG_DIRS for older versions for compatibility
+            (topicPartition, if (config.interBrokerProtocolVersion >= KAFKA_2_7_IV0) Errors.NOT_LEADER_OR_FOLLOWER else Errors.REPLICA_NOT_AVAILABLE)
```

Comment:
Hmm.. Wouldn't it make more sense to tie this to a version bump to this protocol rather than the IBP?

Reply:
Yeah, this was a bad suggestion on my part as I didn't realize that we were using this from clients/tools versus internally. Maybe we should revert to using REPLICA_NOT_AVAILABLE here.

Reply:
Reverted.